### ISSUE TITLE
crash when displaying pointclouds in release mode

### ISSUE THREAD

**Issue by tompollok** [PointCloudLibrary/pcl#1993]
:warning: This is a issue tracker, please use our mailing list for questions: www.pcl-users.org. :warning: 

I am displaying a point cloud that is reconstructed from a stereo images frame by frame.

the first 1-3 iterations are displayed without a crash, but then i crashes in release mode only while it never happens in debug mode.

I use 
pcl::visualization::PCLVisualizer* viewer; as a viewer and 
pcl::PointCloud<pcl::PointXYZRGB>::Ptr m_ptrCloud; as the data container.

the viewer gets initialzied in the constructor of my display class and the m_ptrCloud gets created every time i want to update my point cloud given the new reconstructed results.

new pointcloud data gets displayed by 

if(!viewer->updatePointCloud(m_ptrCloud, "cloud"))
{
     viewer->addPointCloud(m_ptrCloud, "cloud");
}

The error message is as follows:

![grafik](https://user-images.githubusercontent.com/4457821/30323100-367a08e6-97bc-11e7-96d7-95703f2a4d5c.png)

Does anyone have an idea?

## Your Environment
* Operating System and version: Windows 10
* Compiler: MSVC2015 64 bit
* PCL Version: 1.8.1 with VTK 8

## Expected Behavior

Should not crash as in debug mode.

## Current Behavior

crash in release mode after adding the second or third pointcloud
## Possible Solution

i dont know

## Context

displaying stereo camera reconstruction results in real time, or at least as quickly as the processing/upload pipeline works.

**Comment by taketwo** [comment]
Please post an MWE to reproduce the problem. This will allow us to narrow down the scope to particular platform, VTK version, etc.

**Comment by tompollok** [comment]
I tried to create a minimal example and im not able to reproduce this bug in a minimal example. I tested it using the same dll's that were used when the application crashed. do you have an advice what i could possibly do?

--

I tried some desperate std::cout debugging in release mode. The application crashes when i call

viewer->updatePointCloud(m_ptrCloud, "cloud")

**Comment by tompollok** [comment]
I tried to create a minimal example, but i failed reproducing the crash. It works as expected in the minimal example :/

I created a processing pipeline that loads stereo images, estimates a disparity map using opencv and created a 3d point cloud from the disparity map. The stereo reconstruction is opencv accelerated.

If i manage to create a minimal example that reproduces the crash, then ill upload it here.

**Comment by tompollok** [comment]
I found out that as we built pcl (1.8.1) and all its dependencies ourself, the vtk 8 library was unfortunately built only in debug. Unfortunately VTK does not append a d or _debug suffix, but make use of a Realse and Debug destintation folder for the built library.

So with vtk 8 compiled in release mode, the application runs perfectly while in the debug mode i get a crash similiar to what i described for when i run the application in release mode with debug VTK.
